sdk/browser/test/helpers/TestmailInbox.ts

```diff
@@ -0,0 +1,59 @@
+import fetch from 'node-fetch'
+import { URL } from 'url'
+import cryptoRandomString from 'crypto-random-string'
+
+const TESTMAIL_API_URL = 'https://api.testmail.app/api/json'
+const TESTMAIL_INBOX_DOMAIN = 'inbox.testmail.app'
+const TESTMAIL_API_KEY = process.env.TESTMAIL_API_KEY
+const TESTMAIL_NAMESPACE = process.env.TESTMAIL_NS
+
+interface InboxOptions {
+  prefix: string
+  suffix: string
+}
+
+interface Email {
+  subject: string
+  body: string
+}
+
+// NOTE: Don't use this helper for mere email generation,
+//       since there are limits for emails sent per month
+//       Use generateEmail() instead
```

Comment:
I do not understand what this comment means.
Does "email" in "for mere email generation" refer to the address or to the message?
When should I use this helper and when should I use `generateEmail`?
Other developers in the future might also have problems with understanding it.

Reply:
We need to avoid using this helper just for email generation, unless we want to receive emails and read them.
It's because `testmail` is a paid product, and allows no more than 10,000 mails per month for our current plan.
So for random email generation (and we don't care about the mails) we can use `generateEmail()`.
If you understood me, it would be nice if you helped me to shape this comment into a useful state.

Reply:
I think I start to understand it now.

But if no mail messages are sent, then it does not affect the remaining quota.

So we should only avoid using TestmailInbox in cases when:
* We need an email address; and
* Some messages will be sent to this address; and
* We don't care about these messages and will never read them.

Right?

I'm not sure it is such a common scenario for us to guard against it.

Reply:
If I'm not mistaken, email is always sent in `signIn`/`signUp` methods, that are also tested in `NetworkMember.test.ts`. I haven't gone deep to actually test that, but I just wanted to point out that I'm discouraging usage if this helper (and testmail in general) for mere email generation. It doesn't make sense either way, since this service should only be used for testing emails.

Reply:
> If I'm not mistaken, email is always sent in signIn/signUp methods, that are also tested in NetworkMember.test.ts.

If a new user is signed up by email address, then amazon does indeed send the message. However, in that case we would probably also need to get the confirmation code in order to move forward.

What I'm trying to say is:
* If we need to get the message, we should use TestmailInbox;
* If we just need some email address and no messages are getting sent, we should not use TestmailInbox (just because it was created for an entirely different purpose, as it name reflects); but using it will do no harm;
* And the cases where we need an email address, and some messages are getting sent to it, but we don't really care about their content, are not that common, and even if we use TestmailInbox there, it will not do much harm (maybe it will increase total number of messages by 10% or something like that). Even then, it is probably easier to use TestmailInbox there from the standpoint of code uniformity; and anyway, where else do we get a temporary address? Sending some test messages to testmail and some other to gdwk.in does not seem to be a good idea.

Reply:
Still, I want to separate "testmail" inboxes and some random email generation.
Even 10% is huge amount, I've tried to minimize emails sent as much as I could, so I'm sticking to that - it allows us to keep using the cheaper plan of testmail, and also I don't think we should use a whole service helper just for an email generation.

> Sending some test messages to testmail and some other to gdwk.in does not seem to be a good idea.

We can just use something like `example.com`, it's not a problem.